docs/DeveloperGuide.md

```diff
@@ -775,6 +805,41 @@ testers are expected to do more *exploratory* testing.
 
 ***
 
+### Adding a student
+
+1. Adding a student
+
+    1. Prerequisites: Delete any student named `Alice Lee` before and between test cases. Ensure there is a student named `Alex Yeoh` in the list of students.
+
+    2. **Valid Test Cases**
+
+        1. Add a student with only compulsory fields: `add n/alice lee p/91234567 e/91234567 a/123 Clementi`<br>
+           Expected: New student with name `Alice Lee`, phone and emergency contact number `91234567`,
+           and address `123 Clementi` is added. No level, subject(s) or lesson time(s) are specified.
+
+        2. Add a student with compulsory fields, level, subject and lesson times:
+           `add n/alice lee p/91234567 e/91234567 a/123 Clementi l/s1 na s/math s/physics lt/SUN-12:00-14:00 lt/WED-17:00-19:00`<br>
+           Expected: New student with name `Alice Lee`, phone and emergency contact number `91234567`,
+           address `123 Clementi`, subject tags `S1 NA MATH` and `S1 NA PHYSICS`, and lesson times `WED-17:00-19:00` and `SUN-12:00-14:00` is added.
+
+        3. Other valid test cases: Change the capitalisation and spacing between words for the name, level, subject for
+           any of the above test cases.<br>
+           Expected: The expected result is the same as the unaltered test case.
+
+    3. **Invalid Test Cases**
+
+        1. Add a student with subject without level: `add n/Alice Lee p/91234567 e/91234567 a/123 Clementi s/MATH`<br>
+           Expected: No student is added. Error details shown in the status message. Status bar remains the same.
+       
+        2. Add an existing student: `add n/Alex Yeoh p/91234567 e/91234567 a/123 Clementi`<br>
```

Comment:
Invalid test cases 2 and 3 expected can just put Similar to previous (follow find and delete) since message is the same